### ISSUE TITLE
Telemeter 2.0.0 custom component not working?

### ISSUE THREAD

**Issue by fanaticDavid** [thibmaek/custom-components#11]
- Home Assistant version: `0.89.2`
- Host: `Raspberry Pi 3 B+`
- Home Assistant setup:
  - [ ] Hassbian / Raspbian
  - [x] Docker
  - [ ] Hass.io
  - [ ] Python

I can't seem to get the Telemeter custom_component to work. I've followed the instructions in the README and this is my config:
```yaml
- platform: telemeter
  name: "Telemeter"
  cookie: 'AMCV_[REDACTED]%40AdobeOrg'
  include_offpeak: true
```
In place of `[REDACTED]` was a series of 24 letters and digits. The error I get with this config is this:
```
ERROR (SyncWorker_18) [custom_components.telemeter.sensor] Could not decode received JSON: Expecting value: line 1 column 1 (char 0)
ERROR (MainThread) [homeassistant.components.sensor] telemeter: Error on device update!
Traceback (most recent call last):
  File "/config/custom_components/telemeter/sensor.py", line 144, in make_request
    data = response.json()
  File "/usr/local/lib/python3.7/site-packages/requests/models.py", line 889, in json
    self.content.decode(encoding), **kwargs
  File "/usr/local/lib/python3.7/site-packages/simplejson/__init__.py", line 518, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.7/site-packages/simplejson/decoder.py", line 370, in decode
    obj, end = self.raw_decode(s)
  File "/usr/local/lib/python3.7/site-packages/simplejson/decoder.py", line 400, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/homeassistant/helpers/entity_platform.py", line 248, in _async_add_entity
    await entity.async_device_update(warning=False)
  File "/usr/local/lib/python3.7/site-packages/homeassistant/helpers/entity.py", line 348, in async_device_update
    await self.hass.async_add_executor_job(self.update)
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/telemeter/sensor.py", line 157, in update
    response = self.make_request()
  File "/config/custom_components/telemeter/sensor.py", line 152, in make_request
    _LOGGER.error(data)
UnboundLocalError: local variable 'data' referenced before assignment
```
I then tried to troubleshoot it myself by adding `data = {}` on line 140, but then I got the following error instead:
```
ERROR (MainThread) [homeassistant.components.sensor] telemeter: Error on device update!
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/homeassistant/helpers/entity_platform.py", line 248, in _async_add_entity
    await entity.async_device_update(warning=False)
  File "/usr/local/lib/python3.7/site-packages/homeassistant/helpers/entity.py", line 348, in async_device_update
    await self.hass.async_add_executor_job(self.update)
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/telemeter/sensor.py", line 165, in update
    response['usage_peak'] + response['usage_offpeak'])
TypeError: 'NoneType' object is not subscriptable
```

Disclaimer: I am a new user of this custom component so I have never been able to get this to work before.

I only found out about this via a recent retweet by the Home Assistant account. I was eager to start using it, but I was in the middle of migrating from a RPi 3 to an RPi 3 B+, and I had yet to upgrade from 0.87.1 to 0.89.2. Both of those tasks have been completed, and that's when I started trying to use this custom component.
I suspect something might be wrong with the information I'm supplying as the `cookie`, and the errors I'm seeing are a result of that.

**Comment by fanaticDavid** [comment]
Okay so...in a last attempt to figure out if my `cookie` info is indeed the problem, I supplied the entire content of the cookie...and it works!

Maybe there are ways for the script to handle such an error more gracefully, but I am going to close this issue 😉

**Comment by thibmaek** [comment]
Yeah, I've had the same issue in the past. You need to supply the full cookie unfortunately.

This whole component is a hack since there is no easy way to publically access Telenet's OCAPI api. I'll see what I can do to handle this kind of error better.

**Comment by fanaticDavid** [comment]
Yes, Telenet could make this a little easier, but they probably won't. I do appreciate having this hack, though!

I've used a different script before (found [here](https://gist.github.com/VuokkoVuorinnen/38fa003f2da859464ed814e0a5d04d5d)) and combined it with a couple of command line sensors to retrieve the data I wanted, but it stopped working quite some time ago. For a while, I was able to repair it myself but eventually it surpassed my limited coding skills.
